### PR TITLE
Use caxa for binary compilation

### PR DESCRIPTION
**Pull request by maxb2**
This uses [caxa](https://github.com/leafac/caxa) for binary compilation and removes [pkg.](https://github.com/vercel/pkg) 

~~ARM compilation is currently blocked by leafac/caxa#4.~~

See [here](https://github.com/maxb2/dungeon-revealer/releases/tag/v1.14.1-test34) for a test release.

### CODE REVIEW

**Review by maxb2 — COMMENTED** [review]
I updated the dependencies with `npm@7` to resolve the `lockfileVersion` difference

File: package.json [hunk 1]
```diff
@@ -17,10 +17,7 @@
     "build:frontend": "node scripts/copy-monaco-editor-files.js && npm run relay-compiler && vite build --base=./",
     "build:backend": "tsc --project server/tsconfig.json",
     "build": "npm run build:frontend && npm run build:backend",
-    "compile:armv7": "pkg . --out-path ./bin/ --targets node12.18.1-linux-armv7",
-    "compile:arm64": "pkg . --out-path ./bin/ --targets node12.18.1-linux-arm64",
-    "compile:x64": "pkg . --out-path ./bin/ --targets node12.18.1-win-x64,node12.18.1-macos-x64,node12.18.1-linux-x64",
-    "compile": "pkg . --out-path ./bin/ --target host",
+    "compile": "npx caxa -i . -o bin/dungeon-revealer-bin -- \"{{caxa}}/node_modules/.bin/node\" \"{{caxa}}/server-build/index.js\"",
```
Comment (maxb2) [inline package.json:20]: I feel that the output name should be something better than `dungeon-revealer-bin`

**Comment by maxb2** [comment]
The binary sizes are quite big, ~120MB. The binaries are ~40MB when built from a138ec7, [associated release](https://github.com/maxb2/dungeon-revealer/releases/tag/v1.14.1-test12).

**Comment by n1ru4l** [comment]
> The binary sizes are quite big, ~120MB. The binaries are ~40MB when built from [a138ec7](https://github.com/dungeon-revealer/dungeon-revealer/commit/a138ec7005b9e8341eaf92d3f9ce065905d31221), [associated release](https://github.com/maxb2/dungeon-revealer/releases/tag/v1.14.1-test12). I need to investigate this a bit more.

probably dev deps or other unwanted files

**Comment by maxb2** [comment]
@n1ru4l This is finally ready for review!

**Review by n1ru4l — APPROVED** [review]
Finally 🥳 

we can fix caxa later.

**Comment by maxb2** [comment]
@leafac we're officially using caxa now!

**Comment by leafac** [comment]
🙌 Awesome!

@n1ru4l

> we can fix caxa later.

What needs to get fixed?

**Comment by n1ru4l** [comment]
@leafac It seems like the include option paired woth exclude  is a bit weird and copies a lot of unwanted stuff such as .git directories. For now we used patch-package for changing the exclude option to be an include option 😂

**Comment by maxb2** [comment]
@leafac see leafac/caxa#27

**Pull request merged**